Add pan width and position using the new helpers

paulfd opened the pull request:
Corrected some bugs in the modulation of width and position in the course of the implementation.

paulfd committed:
Multiply pan on the channels

paulfd committed:
Disable the SIMD accelerator on pan

paulfd committed:
Changed to the pan helper

paulfd committed:
Changed the way the pan CC is computed

paulfd committed:
Added a width snippet that uses the pan LUT

paulfd committed:
Changed the M/S processing to use the pan/width helpers

paulfd committed:
Position and width correctly map their own cc modulation

paulfd committed:
Use the CC switch on the initial width and position with proper scaling

paulfd committed:
Remove unused spans

jpcima commented on src/sfizz/Voice.cpp:
```diff
@@ -63,19 +63,19 @@ void sfz::Voice::startVoice(Region* region, int delay, int number, uint8_t value
     auto pan { basePan };
     if (region->panCC)
-        pan += normalizeCC(midiState.getCCValue(region->panCC->first)) * normalizeNegativePercents(region->panCC->second);
+        pan += normalizeCC(midiState.getCCValue(region->panCC->first)) * region->panCC->second / 100.0f;
```
division is slower, better to multiply in cases of a constant.
In gcc, `fast-math` will resolve this to multiplication, but maybe not so on MS where I believe the fast-math equivalent is not available.

paulfd committed:
Remove a helper; no clamping for normalizePercent

paulfd committed:
Use the helper with a multiplication

jpcima commented:
I plotted mid and side coeffs, resulting in this data. This is correct?

![Capture du 2020-02-10 00-26-09](https://user-images.githubusercontent.com/17614485/74112337-0d662200-4b9c-11ea-90ae-6616e9d2fd28.png)

Logarithmic

![Capture du 2020-02-10 00-30-43](https://user-images.githubusercontent.com/17614485/74112423-d0e6f600-4b9c-11ea-9c2e-c063a376bbf7.png)

paulfd committed:
Missed a normalization

paulfd commented:
Nope, s***. There it should be better, I forgot the pan table has pi/2 and not pi/4. It should be between ~0.707  and 1 for the mid and ~-0.707 and .707 for the side.

paulfd committed:
Removed an unused span

jpcima commented:
New: linear and log

![Capture du 2020-02-10 00-52-04](https://user-images.githubusercontent.com/17614485/74112762-9894e700-4b9f-11ea-9b35-570cfc56248b.png)
![Capture du 2020-02-10 00-52-19](https://user-images.githubusercontent.com/17614485/74112766-9f235e80-4b9f-11ea-8ae4-eb7cc549bfb6.png)

paulfd committed:
Merge branch 'develop' into width-position

paulfd commented:
OK I hope I did not screw things up with the git thingy. Should be OK apparently?

jpcima commented on src/sfizz/Voice.cpp:
```diff
@@ -364,33 +354,28 @@ void sfz::Voice::processStereo(AudioSpan<float> buffer) noexcept
+
+    // Add const aliases to be slightly more readable
+    const auto midBuffer = leftBuffer;
+    const auto sideBuffer = rightBuffer;
```
shouldn't these 2 be defined as sum and difference of L/R channels?
mid=L+R
side=L-R

paulfd replied:
So the mixing happened a couple lines above but it may have been reversed. I corrected to try to be a bit clearer, does it make sense to you now?

paulfd committed:
Changes to the mono and stereo processes
- Corrected possible issues in the computation
- Removed an unnecessary copy in the mono process
- Trying to make the processes clearer

jpcima committed:
Add the test program for stereo

jpcima commented:
It's incorrect. pan=0 width=100 should reproduce the signal identically.

Sine 220Hz on the left, 880Hz on the right.

![Capture du 2020-02-10 16-27-22](https://user-images.githubusercontent.com/17614485/74163371-96c03780-4c22-11ea-9694-5e8963abf098.png)

paulfd commented:
There's a problem with the mixing constants it seems, the residual sines ~15-20db spaced?

jpcima commented:
Here, the width formula rewritten.
[stereo.dsp.gz](https://github.com/sfztools/sfizz/files/4182072/stereo.dsp.gz)

It's just written out of personal intuition, not off the paper.
It does exactly what the opcode says: mono at 0, identical at +1, reversed at -1

paulfd committed:
Changed the signature of the pan/width/position snippets

paulfd committed:
Added benchmarks and tests for width/pan

paulfd committed:
Changed the width helpers ; need SIMD version still

paulfd committed:
Changed the width/position process again

paulfd committed:
Updated the stereo demo

paulfd committed:
Add the SSE version (to test but it's not faster)
